onnxruntime/core/session/inference_session.cc

```diff
@@ -1130,51 +1158,73 @@ common::Status InferenceSession::Initialize() {
     // Register 2nd registries into KernelRegistryManager.
     ORT_RETURN_IF_ERROR_SESSIONID_(kernel_registry_manager_.RegisterKernels(execution_providers_));
 
-#if !defined(ORT_MINIMAL_BUILD)
-    // add predefined transformers
-    AddPredefinedTransformers(graph_transformation_mgr_, session_options_.graph_optimization_level,
-                              transformers_to_enable_);
-
-    // apply any transformations to the main graph and any subgraphs
-    ORT_RETURN_IF_ERROR_SESSIONID_(TransformGraph(graph, graph_transformation_mgr_,
-                                                  execution_providers_, kernel_registry_manager_,
-                                                  insert_cast_transformer_,
-                                                  *session_state_));
-
-    // now that all the transforms are done, call Resolve on the main graph. this will recurse into the subgraphs.
-    ORT_RETURN_IF_ERROR_SESSIONID_(graph.Resolve());
+    bool loading_ort_format = !ort_format_model_bytes_.empty();
+    bool saving_model = !session_options_.optimized_model_filepath.empty();
+    bool saving_ort_format = false;
+    if (saving_model) {
+      std::string model_type = session_options_.GetConfigOrDefault(kOrtSessionOptionsConfigSaveModelFormat, "");
```

Comment:
Do we want to support saving model when loading_ort_format?

Reply:
Probably not. You could run the ORT model partitioning and compile nodes which results in something that can't be saved. 

I will address in a separate PR though. There was a github issue this week where someone was attempting to save a model that had compiled TensorRT nodes, so we can just do a more general check that there are no compiled kernels and disallow save in that case. 

With that check, it doesn't matter if someone attempts to load and save an ORT format model as it won't be changed given the only thing PartitionOrtFormatModel can do is compile kernels.

---
In reply to: [522558762](https://github.com/microsoft/onnxruntime/pull/5776#discussion_r522558762) [](ancestors = 522558762)